File: 2024/March/Python/bubblesort.py
#!/usr/bin/python3
#
# Mergesort
#

import random
from time import process_time

def bubbleSort(alist):

    for i in range(len(alist)):
        trocou = False
        for j in range(len(alist)-i-1):
            if alist[j] > alist[j+1]:
                alist[j], alist[j+1] = alist[j+1], alist[j]
                trocou = True
        if not trocou:
           break

#alist = []
#for x in range(1000):
#    alist.append(random.randint(0,10000))

#alist = [54,26,93,17,77,31,44,55,20]

MAX = 10000
data = [random.randint(0,MAX*10) for x in range(MAX)]

start = process_time()
bubbleSort(data)
end = process_time()
t = end-start

print(data)
print(f"Tempo para ordenar {MAX} elementos: {t} s")

File: 2024/March/Og_Java/Bubblesort.java
import java.util.Random;

public class Bubblesort {

    public void sort(int[] values) {
        for(int i=0; i<values.length; i++) {
            boolean trocou = false;
            for(int j=0; j<values.length-i-1; j++) {
                if(values[j] > values[j+1]) {
                    int temp = values[j];
                    values[j] = values[j+1];
                    values[j+1] = temp;
                    trocou = true;
                }
            }
            if(!trocou) break;
        }
    }

    public static void main(String args[]) {
        Random r = new Random();
        int MAX = 10000;
        int[] data = new int[MAX];
        for(int i=0; i<data.length; i++)
            data[i] = r.nextInt(data.length*10);

        long start = System.nanoTime();
        Bubblesort bs = new Bubblesort();
        bs.sort(data);
        long end = System.nanoTime();
        //for(int i=0; i<data.length; i++)
        //    System.out.print(data[i]+" ");
        //System.out.println();
        System.out.printf("Tempo para ordenar %d elementos: %d ns\n",MAX,end-start);
    }
}
